R/bayesianpredictionperformance.R

```diff
@@ -0,0 +1,259 @@
+
+  rmPlot$plotObject <- p
+
+  return()
+}
+.metamiscAcPlot                <- function(jaspResults, options, dataset, ready) {
+
+  if (!is.null(jaspResults[["acPlot"]]))
+    return()
+
+  imgHeight  <- 500
+  imgWidth   <- 250 * options[["chains"]]
+
+  acPlot   <- createJaspPlot(title = gettext("Autocorrelations plot"), width = imgWidth, height = imgHeight)
+  acPlot$position <- 6
+  acPlot$dependOn(c("diagnosticsAcPlot", c(.metamiscDependencies, .metamiscDependenciesBayesian)))
+  jaspResults[["acPlot"]] <- acPlot
+
+  if (!ready)
+    return()
+
+  p <- try(metamisc::acplot(jaspResults[["model"]][["object"]]))
+
+  if (isTryError(p)) {
+    acPlot$setError(.extractErrorMessage(p))
+    return()
+  }
+
+  p <- jaspGraphs::themeJasp(p, sides = "bl") + ggplot2::xlab(gettext("Lag")) + ggplot2::ylab(gettext("Autocorrelation"))
+
+  acPlot$plotObject <- p
+
+  return()
+}
+.metamiscGelmanRubinPlot       <- function(jaspResults, options, dataset, ready) {
+
+  if (!is.null(jaspResults[["gRPlot"]]))
+    return()
+
+  imgHeight  <- 250
+  imgWidth   <- 600
+
+  gRPlot   <- createJaspPlot(title = gettext("Autocorrelations plot"), width = imgWidth, height = imgHeight)
+  gRPlot$position <- 7
+  gRPlot$dependOn(c("gelmanRubinPlot", c(.metamiscDependencies, .metamiscDependenciesBayesian)))
+  jaspResults[["gRPlot"]] <- gRPlot
+
+  if (!ready)
+    return()
+
+  p <- try(metamisc:::gelmanplot(jaspResults[["model"]][["object"]]))
+
+  if (isTryError(p)) {
+    gRPlot$setError(.extractErrorMessage(p))
+    return()
+  }
+
+  p <- jaspGraphs::themeJasp(p, sides = "bl") + ggplot2::xlab(gettext("Last iteration in chain")) + ggplot2::ylab(gettext("Shrink factor"))
+
+  gRPlot$plotObject <- p
+  return()
+}
+.metamiscOmitNAs               <- function(dataset, options) {
+
+  dataset$computable <- FALSE
+
+  if (options[["inputMeasure"]] != "" && options[["inputSE"]] != "")
+    dataset$computable <- ifelse(!is.na(dataset[, options[["inputMeasure"]]]) & !is.na(dataset[, options[["inputSE"]]]), TRUE, dataset$computable)
+
+  if (options[["inputMeasure"]] != "" && sum(unlist(options[["inputCI"]]) != "") == 2)
+    dataset$computable <- ifelse(!is.na(dataset[,options[["inputMeasure"]]]) & !is.na(dataset[, options[["inputCI"]][[1]][1]]) & !is.na(dataset[, options[["inputCI"]][[1]][2]]), TRUE, dataset$computable)
+
+  if (options[["measure"]] == "cstat" && options[["inputMeasure"]] != "" && options[["inputN"]] != "" && options[["inputO"]] != "")
+    dataset$computable <- ifelse(!is.na(dataset[, options[["inputMeasure"]]]) & !is.na(dataset[, options[["inputN"]]]) & !is.na(dataset[, options[["inputO"]]]), TRUE, dataset$computable)
+
+  if (options[["measure"]] == "OE" && options[["inputE"]] != "" && options[["inputO"]] != "")
+    dataset$computable <- ifelse(!is.na(dataset[, options[["inputE"]]]) & !is.na(dataset[, options[["inputO"]]]), TRUE, dataset$computable)
+
+  dataset <- dataset[dataset$computable, ]
```

Comment:
at the end here you'll always have dataset$computable in there? why not:

```suggestion
  
  computable <- rep(FALSE, nrow(dataset))

  if (options[["inputMeasure"]] != "" && options[["inputSE"]] != "")
    computable <- computable | (!is.na(dataset[, options[["inputMeasure"]]]) & !is.na(dataset[, options[["inputSE"]]])

  if (options[["inputMeasure"]] != "" && sum(unlist(options[["inputCI"]]) != "") == 2)
    computable <- computable | (!is.na(dataset[,options[["inputMeasure"]]]) & !is.na(dataset[, options[["inputCI"]][[1]][1]]) & !is.na(dataset[, options[["inputCI"]][[1]][2]])

  if (options[["measure"]] == "cstat" && options[["inputMeasure"]] != "" && options[["inputN"]] != "" && options[["inputO"]] != "")
    computable <- computable | (!is.na(dataset[, options[["inputMeasure"]]]) & !is.na(dataset[, options[["inputN"]]]) & !is.na(dataset[, options[["inputO"]]]))

  if (options[["measure"]] == "OE" && options[["inputE"]] != "" && options[["inputO"]] != "")
    computable <- computable | (!is.na(dataset[, options[["inputE"]]]) & !is.na(dataset[, options[["inputO"]]]))

  dataset <- dataset[computable, ]
```
so basically, join the logicals with `computable | (conditions)` rather than `ifelse(condition, TRUE, computable)`.